charmhelpers/contrib/openstack/context.py

```diff
@@ -352,10 +352,101 @@ def _setup_pki_cache(self):
             return cachedir
         return None
 
+    def _get_ubuntu_pkg_name(self, python_name):
+        """
+        We assume Python3 is the main interpreter and python3-keystonemiddleware
+        is the main package for the transitional period. python-keystonemiddle is
+        here to support possible future
+        (Python 3 and python3-keystonemiddleware->python-keystonemiddleware ?)
+        and the past
+        (Python 2 and python-keystonemiddleware)
+        """
+
+        pkg = 'python3-'
+        if filter_installed_packages((pkg + python_name,)):
+            pkg = 'python-' + python_name
+
+        return pkg
+
+    def _get_keystone_authtoken_ctxt(self, ctxt, keystonemiddleware_os_rel):
+        """
+        Re-constructed from former template 'section-keystone-auth-mitaka'
+        """
+        config_items = []
+
+        if CompareOpenStackReleases(keystonemiddleware_os_rel) >= 'mitaka':
+            config_items += (
+                ('auth_type', 'password'),
+            )
+        else:
+            config_items += (
+                ('auth_plugin', 'password'),
+            )
+
+        api_version = ctxt.get('api_version')
+
+        auth_uri = "{service_protocol}://{service_host}:{service_port}" \
+                   .format(**ctxt)
+
+        auth_url = "{auth_protocol}://{auth_host}:{auth_port}" \
+                   .format(**ctxt)
+
+        if float(api_version) >= 3.0:
```

Comment:
Queens is v3 only so we can remove this conditional

Reply:
If we only support v3, then I think it would be good to remove extra conditions from keystone-authtoken-mitaka as well to clean up.